airbyte-integrations/connectors/source-amazon-ads/source_amazon_ads/streams/sponsored_display.py

```diff
@@ -12,11 +12,17 @@ class SponsoredDisplayCampaigns(SubProfilesStream):
     https://advertising.amazon.com/API/docs/en-us/sponsored-display/3-0/openapi#/Campaigns
     """
 
+    def __init__(self, *args, **kvargs):
+        state = kvargs.get("config", {}).get("stateFilter")
+        self.state_filter = ",".join(state) if state else ""
```

Comment:
In the constructor, store the instance variable and do all the formatting in the `request_parameters` method.

```suggestion
        self.state_filter = kwargs.get("config", {}).get("state_filter")
```